cpp/oneapi/dal/algo/decision_forest/backend/gpu/train_splitter_impl_dpc.cpp

```diff
@@ -634,6 +727,26 @@ train_best_split_impl<Float, Bin, Index, Task, use_private_mem>::compute_best_sp
     const sycl::nd_range<2> nd_range =
         bk::make_multiple_nd_range_2d({ local_size, node_count }, { local_size, 1 });
 
+    auto ftr_random_select = pr::ndarray<Index, 1>::empty({ selected_ftr_count });
+    auto ftr_random_select_ptr = ftr_random_select.get_mutable_data();
+
+    if (ctx.splitter_mode_value == splitter_mode::random) {
+        pr::engine_collection collection(1, ctx.seed);
+        std::vector<pr::engine> engine_arr = collection([&](std::size_t i, std::size_t& j) {
+            j = i;
+        });
+        pr::rng<Index> rn_gen;
+        rn_gen.uniform( // Select bin treshold randomly
+            selected_ftr_count,
+            ftr_random_select_ptr,
+            engine_arr[0].get_state(),
+            0,
+            max_bin_count_among_ftrs);
```

Comment:
It seems there is no need in a vector of random number generator (RNG) states here. As there is no need in parallelism across the histogram's bins below (cgh.parallel_for at line #755). As the only 1 split is processed for each feature in case of extra trees.